internal/encoding/merge/parquet.go

```diff
@@ -0,0 +1,330 @@
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_FLOAT)
+		fieldHandler = floatHandler
+	case "double":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
+		fieldHandler = doubleHandler
+	case "int":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		fieldHandler = intHandler(64)
+	case "json":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.JSON = &parquet.JsonType{}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_JSON)
+		fieldHandler = jsonHandler
+	default:
+		return nil, nil, fmt.Errorf("unsupported type %q", typ)
+	}
+
+	fieldHandler = optionalHandler(fieldHandler)
+
+	return col, fieldHandler, nil
+}
+
+func byteArrayHandler(s interface{}) (interface{}, error) {
+	switch v := s.(type){
+	case []byte:
+		return v, nil
+	case string:
+		return []byte(v), nil
+	case json.RawMessage:
+		return []byte(v), nil
+	default:
+		return []byte(fmt.Sprintf("%v", s)), nil
+	}
+}
+
+func booleanHandler(s interface{}) (interface{}, error) {
+	switch v := s.(type){
+	case bool:
+		return v, nil
+	case string:
+		return strconv.ParseBool(v)
+	default:
+		return nil, fmt.Errorf("Unknown format")
+	}
+}
+
+func uintHandler(bitSize int) func(interface{}) (interface{}, error) {
+	return func(s interface{}) (interface{}, error) {
+		localString := fmt.Sprintf("%v", s)
+		i, err := strconv.ParseUint(localString, 10, bitSize)
+		if err != nil {
+			return nil, err
+		}
+		switch bitSize {
+		case 8, 16, 32:
+			return uint32(i), nil
+		case 64:
+			return i, nil
+		default:
+			return nil, fmt.Errorf("invalid bit size %d", bitSize)
+		}
+	}
+}
+
+func intHandler(bitSize int) func(interface{}) (interface{}, error) {
+
+	return func(s interface{}) (interface{}, error) {
+		switch v := s.(type) {
+		case int:
+			return intHandlerHelper(bitSize, int64(v))
+		case int64:
+			return intHandlerHelper(bitSize, v)
+		case int32:
+			return intHandlerHelper(bitSize, int64(v))
+		case string:
+			var intVal, err = strconv.ParseInt(v, 10, bitSize)
+			if err != nil {
+				return nil, err
+			}
+
+			return intHandlerHelper(bitSize, intVal)
+		default:
+			return nil, fmt.Errorf("invalid bit size %d", bitSize)
+		}
+	}
+}
+
+func intHandlerHelper(bitSize int, rawValue int64) (interface{}, error) {
+	switch bitSize {
+	case 8, 16, 32:
+		return int32(rawValue), nil
+	case 64:
+		return rawValue, nil
+	default:
+		return nil, fmt.Errorf("invalid bit size %d", bitSize)
+	}
+}
+
+func floatHandler(s interface{}) (interface{}, error) {
+	switch v := s.(type){
+	case float32:
+		return v, nil
+	default:
+		return nil, fmt.Errorf("Error in parsing")
```

Comment:
Update error message: `fmt.Errorf("toparquet: unable to parse %T as float32", s)`